app/controllers/queues_controller.rb

```diff
@@ -1,24 +1,58 @@
 class QueuesController < ApplicationController
   def spam
-    @request = RescueRequest.find_by(needs_spam_check: true)
+    @request = RescueRequest.find_by(needs_spam_check: nil)
+  end
+
+  def spam_spam
+    request = RescueRequest.find(params[:request_id])
+    request.update(needs_spam_check: false)
+    request.review_tasks.create(user_id: current_user.id, review_type: "spam", outcome: "spam")
+    redirect_to :spam_queue
+  end
+
+  def spam_not
+    request = RescueRequest.find(params[:request_id])
+    request.update(needs_spam_check: false)
+    request.review_tasks.create(user_id: current_user.id, review_type: "spam", outcome: "not spam")
+    redirect_to :spam_queue
   end
 
   def validation
     @request = RescueRequest.find_by(needs_validation: true)
   end
 
   def deduping
-    @request = RescueRequest.find_by(needs_deduping: true)
-    criteria = @request.attributes.map { |col, val| " #{col} LIKE #{val}" unless val.nil? || val.empty? }.reject { |i| i.nil? || i.empty? }.join(" OR ")
+    @request = RescueRequest.find_by(needs_deduping: nil)
+    criteria = @request.attributes.select {|col, _val| %w[lat long twitter phone email medical conditions extra_details street_address].include? col }.map { |col, val| " #{col} LIKE #{ActiveRecord::Base.connection.quote(val.to_s)}" unless val.nil? || val.to_s.empty? }.reject { |i| i.nil? || i.empty? }.join(" OR ")
```

Comment:
This line needs to... not. It's too long and doing too much - I'm all for one-liners, but there's a limit to how much you should really be doing in one line. This probably needs breaking out into smaller parts, which will also make it easier to debug.

Reply:
Should be fixed with the latest commit.